### ISSUE TITLE
Setting Expires headers

### ISSUE THREAD

**Issue by maxrosecollins** [AssetSync/asset_sync#308]
I noticed `issue 39` fixed the issue for setting the expires HTTP header to the far-future. I don't know is amazon updated something but it doesnt work any more and none of my headers are being set.

**Comment by sbeam** [comment]
same issue here on S3/Cloudfront; I think it might have to do with sprockets not including the gzip files anymore, see #304. We are using the patch included there but the gzipped assets are not triggering the setting of the Expires header when uploaded. There is a patch in https://github.com/rumblelabs/asset_sync/pull/223 that might address this.

```
$ curl -I http://xyz.cloudfront.net/assets/application-7f4d06566e2bcde97fecbbde60081bd6e2dcbe8f412a8105660c126a344e3bfb.css
HTTP/1.1 200 OK
Content-Type: text/css
Content-Length: 6772
Connection: keep-alive
Date: Mon, 06 Jul 2015 04:04:43 GMT
Content-Encoding: gzip
Last-Modified: Tue, 30 Jun 2015 14:50:49 GMT
ETag: "befa12908aa927161b6d9ac470d63054"
Accept-Ranges: bytes
Server: AmazonS3
Age: 45310
X-Cache: Hit from cloudfront
Via: 1.1 97425b66a3749ba768ba59108c1da79a.cloudfront.net (CloudFront)
```

**Comment by lime** [comment]
This seems to be caused by AssetSync assuming the hash to be 32 characters long, while the current hash used is 64.

Also, this has apparently already been fixed in #315, but not yet released.

**Comment by maxrosecollins** [comment]
Okay, great news!

**Comment by PikachuEXE** [comment]
I have no permission to release, so...
Lets wait?

**Comment by lime** [comment]
Yeah, I just read #316 and it sounds like @davidjrice would need to give out release rights for that to happen, right? I'll install from master until then.

**Comment by lime** [comment]
Okay, so the 64 char issue was solved in #315, but `.gz` assets are indeed not handled since #223 hasn't been merged. I could do a rebase of the fix on the current master.

**Comment by PikachuEXE** [comment]
Please do so. Thanks!

**Comment by PikachuEXE** [comment]
Please try `master` since #329 merged
And let us know if this issue is fixed

**Comment by davidjrice** [comment]
@PikachuEXE good work. Would you like release rights?

@lime likewise!

**Comment by PikachuEXE** [comment]
@davidjrice 
Sure <3

I just realize the last release was made in 2014 (O.O)
Maybe you want to make a new one soon? (So I can follow the release style)

**Comment by lime** [comment]
@davidjrice I'm not _that_ familiar with the codebase yet, so I wouldn't dare to make any big judgment calls. But I'm happy to help out as much as I can. :)

**Comment by johansmitsnl** [comment]
+1 for an release of the new features and cache control

**Comment by johan-smits** [comment]
@davidjrice Looking forward of a new release with this feature.
Is a release blocked for some reason?

**Comment by PikachuEXE** [comment]
#329 should be released in `1.2.0` (a recent release)
See https://github.com/AssetSync/asset_sync/blob/master/CHANGELOG.md

**Comment by lime** [comment]
This should indeed be released now since `1.2.0`.

I'll close the issue. If anyone notices problems with the cache headers after updating, please do comment or open an issue. :)
